### PR TITLE
ParseEmailFiles - added code for multiple mime encoding

### PR DESCRIPTION
**Pull request by tallieber**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [39653](https://github.com/demisto/etc/issues/39653)

## Description
1. Docker image was added to yml file on 1.4.5 version. This changed from the default docker -`demisto/python:1.3-alpine` to `demisto/python:2.7.18.20958`. This cause a change when reading the message from the given path.

2. Subject includes 2 encoding parts - added a functionality to go over them

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by tallieber** [comment]
@bakatzir I'm getting this error when I change the docker image to the default file`[DO100] - The current docker image in the yml file is the default one: demisto/python:1.3-alpine` 
And when I remove this docker - it raised error that yml file must include docker.

**Review by bakatzir — APPROVED** [review]
I'm good with the mime multi encoded words decoder addition. also with the dockerimage change. @yuvalbenshalom please have a look as well. if LGTM, force-merge.

File: Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.py [hunk 1]
```diff
@@ -3387,27 +3387,47 @@ def get_utf_string(text, field):
...
+                        # ends with "?=", and has two "?"s in between.
+                        start_encoding_index = encode_decode_phrase.index("=?")
+                        end_encoding_index = encode_decode_phrase.index("?=") + 2
+                        # index return the index where "?=" starts, need to include it on the substring
+                        encoded_substring = encode_decode_phrase[start_encoding_index:end_encoding_index]
+                        decoded_substring = mime_decode(encoded_substring)
+                        encode_decode_phrase = encode_decode_phrase[:start_encoding_index] + decoded_substring + \
+                                               encode_decode_phrase[end_encoding_index:]  # noqa: E127
+                        word_mime_encoded = MIME_ENCODED_WORD.search(encode_decode_phrase)
+                    return encode_decode_phrase
```
Comment (yuvalbenshalom) [inline Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.py:3429]: looks good, please just fix the mix of use in in " and '. use only ' for strings.

**Review by yuvalbenshalom — APPROVED** [review]
@tallieber see my comment, and fix the conflict, and I will force merge.

File: Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.py (hunk 1 above)
Comment (tallieber) [reply Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.py]: It is ready for merge

File: Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.py [hunk 2]
```diff
@@ -181,21 +181,32 @@ def PtypString8(data_value):
...
                 if enc != 'ascii':
                     if enc.lower() == 'windows-1252' and res['confidence'] < 0.9:
                         demisto.debug('encoding detection confidence below threshold {}, '
                                       'switching encoding to "windows-1250"'.format(res))
-                        enc = 'windows-1250'
-                    data_value = data_value.decode(enc, errors='ignore').replace('\x00', '')
-                elif '\x00' not in data_value:
-                    data_value = data_value.decode("ascii", errors="ignore").replace('\x00', '')
+
+                        enc = DEFAULT_ENCODING if DEFAULT_ENCODING else 'windows-1250'
```
Comment (tkatzir) [inline Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.py:193]: Move this above the `demisto.debug` (currently line 190) and use the value of `enc` in the debug message, so it reflects the real encoding you'll use.

Comment (tallieber) [reply Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.py:193]: Agree, will change it accordingly.

File: Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.yml [hunk 3]
```diff
@@ -15,6 +15,12 @@ args:
     contains an emails contains an email). Default depth level is 3. Minimum level
     is 1, if set to 1 the script will parse only the first level email
   name: max_depth
+- required: false
+  name: forced_encoding
+  description: Use only the force encoding when parsing the message. Relevant to msg files only.
+- required: false
+  name: default_encoding
+  description: Use only the default encoding when parsing the message with the detected encoding fails. Relevant to msg files only.
```
Comment (yuvalbenshalom) [inline Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.yml:23]: maybe it is good to add examples of default encodings to make it clear what strings can be added here?

**Review by yuvalbenshalom — APPROVED** [review]
looks good.
see my comment.

**Pull request merged**